controllers/windowsmachine_controller.go

```diff
@@ -337,6 +332,24 @@ func (r *WindowsMachineReconciler) Reconcile(ctx context.Context, request ctrl.R
 	return ctrl.Result{}, nil
 }
 
+// deleteMachineIfAllowed deletes the specified Machine if deletion is allowed.
+// Returns a boolean if the deletion attempt should result in a requeuing.
+func (r *WindowsMachineReconciler) deleteMachineIfAllowed(machine *mapi.Machine) (bool, error) {
+	r.log.Info("deleting machine", "name", machine.GetName())
+	deletionAllowed, err := r.isAllowedDeletion(machine)
+	if err != nil {
+		return false, errors.Wrapf(err, "unable to determine if Machine can be deleted")
+	}
+	if !deletionAllowed {
+		r.log.Info("machine deletion restricted", "maxUnhealthyCount", maxUnhealthyCount)
+		r.recorder.Eventf(machine, core.EventTypeWarning, "MachineDeletionRestricted",
+			"Machine %v deletion restricted as the maximum unhealthy machines can`t exceed %v count",
+			machine.Name, maxUnhealthyCount)
+		return true, nil
+	}
+	return false, r.deleteMachine(machine)
```

Comment:
A positive direction in the execution flow, always reads better.  Consider:

```go
if fooAllowed() {
   doFoo()
   return
}

doNotFoo();
return
```

rather than

```go
if !fooAllowed() {
     doNotFoo();
     return
}
doFoo()
return

```